### PR TITLE
Initial implementation of OAuth

### PR DESCRIPTION
**Pull request by andrey-snowflake**
screen shot 2019-01-16 at 4 38 00 pm
screen shot 2019-01-16 at 4 59 48 pm
screen shot 2019-01-16 at 4 57 16 pm

### CODE REVIEW

File: src/samui/backend/samui/api/oauth.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+def oauth_return():
+    json = request.get_json()
+    code = json.get('code')
+    account = json.get('account')
+
+    response = requests.post(
+        f'https://{account}.snowflakecomputing.com/oauth/token-request',
+        headers={
+            'Content-Type': 'application/x-www-form-urlencoded',
+            'Authorization': 'Basic ' + b64encode(f"{OAUTH_CLIENT_ID}:{OAUTH_SECRET_ID}".encode('utf8')).decode('utf8'),
```
Comment (greg-snowflake) [inline src/samui/backend/samui/api/oauth.py]: Oauth requires that you use basic auth to encode the secret id in the request?

Comment (greg-snowflake) [reply src/samui/backend/samui/api/oauth.py]: talked about this offline, decided it's fine

Comment (andrey-snowflake) [reply src/samui/backend/samui/api/oauth.py]: yeah -- I think this specific piece is part of OAuth standard

**Comment by greg-snowflake** [comment]
this all still looks good

**Pull request merged**